### PR TITLE
IGNITE-14575 Write to DMS must throw error, if client is not in topology

### PR DESCRIPTION
**Pull request by SammyVimes**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java [hunk 1]
```diff
@@ -357,8 +357,9 @@ public static boolean stop(@Nullable String name, boolean cancel,
             grid.starterThread.interrupt();
         }
 
-        if (grid != null && grid.state() == STARTED) {
-            grid.stop(cancel, shutdown);
+        if (grid != null) {
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java:360]: It seems that there is no difference and can not be changed.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java [hunk 2]
```diff
@@ -151,6 +158,46 @@ public void testSingleNode() throws Exception {
...
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testDistributedMetastorageOperationsOnClient() throws Exception {
+        String clientName = "client0";
+
+        String key = "key";
+        String value = "value";
+
+        GridTestUtils.runAsync(() -> startGrid(clientName));
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java:173]: U can use **GridAbstractTest#startClientGrid(int)**

File: modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java [hunk 3]
```diff
@@ -151,6 +158,46 @@ public void testSingleNode() throws Exception {
...
+
+        GridTestUtils.waitForCondition(() -> {
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java:175]: Maybe assertTrue ?

File: modules/zookeeper/src/main/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoveryImpl.java [hunk 4]
```diff
@@ -668,6 +668,11 @@ public boolean knownNode(UUID nodeId) {
     public void sendCustomMessage(DiscoverySpiCustomMessage msg) {
         assert msg != null;
 
+        List<ClusterNode> nodes = rtState.top.topologySnapshot();
+
+        if (nodes.stream().allMatch(ClusterNode::isClient))
```
Comment (tkalkirill) [inline modules/zookeeper/src/main/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoveryImpl.java:673]: Maybe use loop ?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java (hunk 2 above)
Comment (SammyVimes) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStorageTest.java:173]: But I need the name to later get the semi-started node with `IgnitionEx.gridx`

**Pull request closed without merging**